README.md

```diff
@@ -24,30 +23,31 @@ import 'package:flutter/widgets.dart';
 import 'package:nuvigator/nuvigator.dart';
 
 @NuRouter()
-class MyRouter extends BaseRouter {
+class MyRouter extends Router {
 
   @NuRoute()
   ScreenRoute myRoute() => ScreenRoute(
-    builder: (context) => MyScreen(screenContext),
+    builder: (_) => MyScreen(),
   );
 
   @override
-  Map<RouteDef, ScreenRouteBuilder> get screensMap  => _$myScreensMap(this);
+  Map<RouteDef, ScreenRouteBuilder> get screensMap  => _$screensMap;
 }
 
-// Wraps the top most Router into a special Router to be able to handle with DeepLinks and system behaviors. This is a
-// must. Be sure to wrap your top most router into a GlobalRouter.
-final router = GlobalRouter(baseRouter: MyRouter());
-
-Widget build(BuildContext context) {
-  return MaterialApp(
-    builder: Nuvigator(
-      router: router,
-      screenType: materialScreenType,
-      initialRoute: MyRoutes.myRoute,
-    ), 
-  );
+class MyScreen extends StatelessWidget {
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      builder: Nuvigator(
+        router: MyRouter(),
```

Comment:
The library itself does not require that the Router retain state. The only thing it needs is to keep track of the Nuvigator that is its owner, and this is being done by the `initState` and didUpdateWidget lifecycle methods.
Adding the `createRouter` is not really possible because we need it in the constructor to call the `super` passing the `onGenerateRoute` which depends on the Router, so we can't wait until the `initState` to run it =/.
If someone would like to retain the state in the Router we can add a Readme section explain that the user should handle the instance using some state management system (provider, StatefulWidget, or any other).